management/management_request.go

```diff
@@ -116,6 +116,33 @@ func (m *Management) Request(ctx context.Context, method, uri string, payload in
 	return nil
 }
 
+func getWithPagination[Resource interface{}](
+	ctx context.Context,
+	opts []RequestOption,
+	api func(ctx context.Context, opts ...RequestOption) (result []Resource, hasNext bool, err error),
+) ([]Resource, error) {
+	const pageSize = 100
+	var list []Resource
+	var pageNumber int
+	for {
+		opts = append(opts, PerPage(pageSize), Page(pageNumber))
```

Comment:
this will lead to opts growing on each iteration. maybe this would be better?

```suggestion
		realOpts = append(opts, PerPage(pageSize), Page(pageNumber))
```

and then use `realOpts` below on the API call.